### ISSUE TITLE
no jobs executing with limit 1

### ISSUE THREAD

**Issue by mdomsch-seczetta** [deanpcmad/sidekiq-limit_fetch#146]
Midway through processing a many-thousand-item queue with limit=1, potentially one one of 4 different hosts, all new job processing from that queue stopped.  Sidekiq shows no jobs running.  Setting limit=2 on the fly then, one job at a time started processing again. It looks as though a "ticket" to handle a job was taken and not released properly.

Perhaps not coincidentally, around the time this occurred, the app lost connectivity to its MySQL server, and then regained connectivity a short while later.

**Comment by mdomsch-seczetta** [comment]
As a manual workaround, I deleted the redis keys for limit_fetch:busy:queue_name when I knew no jobs were actually busy.

is it worth having some method to check that the busy values are actually still-running jobs? Maybe on a timer that runs every so often (say once a minute) ?

**Comment by mdomsch-seczetta** [comment]
we're seeing this again. Seems to be a race condition not accounted for.

**Comment by deanpcmad** [comment]
What version of Sidekiq are you running? Does the latest change in master fix it?

**Comment by mohideen** [comment]
@deanpcmad I was experiencing a similar issue in our setup with version `3.4.0`, and it was fixed by updating to `4.4.1`.

We have an application in kubernetes environment where the sidekiq was running in a pod and the redis was on a different pod. We noticed that sidekiq stoped processing queue if the redis container within the redis pod got restarted. The queue process was not affected when the redis pod restarted, and it only happened when the container within the pod restarted. Our redis pod is unstable due to disk latency issues causing the container to crash periodically and thereby halting the sidekiq queue processing. And, we had to restart sidekiq to resume the queue processing. After upgrading the sidekiq-limit_fetch to `4.4.1`, the container restart no longer stops queue processing.

Do you know what specific change could have helped resolve this issue? I'm asking because we are facing a similar issue in one of our Java/Stomp messaging application, and the fix here could help us understand the root of the problem.

**Comment by deanpcmad** [comment]
What if you try a slightly older version?

**Comment by mohideen** [comment]
@deanpcmad I tested different versions, and I found the issue was resolved in v4.1.0. I see that there are changes to redis connection handling between these versions: https://github.com/deanpcmad/sidekiq-limit_fetch/compare/v4.0.0...v4.1.0

Can you summarize the intent of the changes?

**Comment by deanpcmad** [comment]
Those commits were from 2018 and not written by me, but from a quick glance it's to do with threads
